extensions/interactions/InteractiveMap/directives/OppiaInteractiveInteractiveMapDirective.js

```diff
@@ -72,61 +96,44 @@ oppia.directive('oppiaInteractiveInteractiveMap', [
             refreshMap();
           });
 
-          // This is required in order to avoid the following bug:
-          //   http://stackoverflow.com/questions/18769287
           var refreshMap = function() {
-            $timeout(function() {
-              google.maps.event.trigger($scope.map, 'resize');
-              $scope.map.setCenter({
+            $scope.mapOptions = {
+              center: {
                 lat: coords[0],
-                lng: coords[1]
-              });
-              if (!$scope.interactionIsActive) {
-                $scope.mapMarkers.push(new google.maps.Marker({
-                  map: $scope.map,
-                  position: new google.maps.LatLng(
-                    $scope.getLastAnswer()[0], $scope.getLastAnswer()[1])
-                }));
-              }
-            }, 100);
-          };
-
-          var coords = $scope.coords || [0, 0];
-          var zoomLevel = parseInt($scope.zoom, 10) || 0;
-          $scope.mapOptions = {
-            center: new google.maps.LatLng(coords[0], coords[1]),
-            zoom: zoomLevel,
-            mapTypeId: google.maps.MapTypeId.ROADMAP,
-            draggable: $scope.interactionIsActive
+                lng: coords[1],
+                zoom: zoomLevel
+              },
+              events: {}
+            };
+            if (!$scope.interactionIsActive) {
+              addNewMarker(scope.getLastAnswer()[0], $scope.getLastAnswer()[1]);
+            }
           };
 
-          $scope.onMouseOver = function() {
+          $scope.$on('leafletDirectiveMap.mouseover', function(event) {
```

Comment:
Drop "event" if it's not being used, ditto below.

Reply:
Done!